tests/drivers/uart/uart_async_api/testcase.yaml

```diff
@@ -81,9 +81,6 @@ tests:
     harness_config:
       fixture: gpio_loopback
     depends_on: gpio
-    platform_allow:
-      - nucleo_f746zg
-      - nucleo_h723zg
```

Comment:
better not change this but add supported platform instread

Reply:
I was trying to see why this was added, what platforms are failing if this is not here, but also I was going to confirm if this part of the yaml file is even working, when I build with these two platforms, the extra Kconfig did not appear, neither did they appear for any of the rt series which is causing the tests to fail, however if I manually add the extra Kconfigs on the command line, the rt series passes.
@dcpleung @danieldegrasse 
What am I missing from the file with this information?